modules/sleep-detector/main.py

```
@@ -282,15 +357,21 @@ def read_records(self):
-                record = cbor2.load(self._file)
-                inner = cbor2.loads(record["data"])
+                data_bytes = _read_raw_record(self._file)
+                if data_bytes is None:
+                    self._last_pos = self._file.tell()
+                    continue  # empty placeholder record
+                inner = cbor2.loads(data_bytes)
+                self._last_pos = self._file.tell()
                 yield inner
             except EOFError:
                 time.sleep(0.5)
             except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)